### PR TITLE
S4659241-GCN

### PR DESCRIPTION
**Pull request by kexin28**
Hi Shakes,
This pull request contains the files relating to implementing the GCN task. The model can achieve over 91% accuracy on test dataset. Some descriptions of the files are below:
README.MD -- Description of the code files and implementation details.
algorithm.py -- Code for building the model.
test.py -- A test script for loading the data. preprocessing the data and training the model, showing the result.
images -- A folder contains images used in the README.MD.

Best Regards,
Kexin Peng
46592419

### CODE REVIEW

**Comment by gayanku** [comment]
- PR on Oct 30
- Detailed commit and PR messages
- Readme is well done with background, results, diagrams.
- multi layer (3) layer GCN, Good accuracy (+90%), Semi supervised split.
- **Valid for NORMAL**

- TSNE looks wrong. You need to use the embeddings from the GCN, just prior to the softmax ( self.layer3 ) and use this as the TSNE input. Using the features with PCA as you have done does not incorporate the graph (edge) information in to the embeddings.
- For better reuse, you can move pre-processing and TSNE code etc out from the driver.  Also use a main function.

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
Header blocks
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage 
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description OK
